client/src/components/related_components/ProductCard.jsx

```diff
@@ -44,17 +70,31 @@ const ProductCard = (props) => {
   };
 
   const getRating = (ratings) => {
-    // console.log('ratings in getrating', ratings);
-    // cocounter = 0;
-    // var total = 0;
-    console.log(Object.entries(ratings));
+    let counter = 0;
+    let total = 0;
+
+    Object.entries(ratings).forEach((set) => {
+      const star = Number(set[0]);
+      const quantity = Number(set[1]);
+      total += star * quantity;
+      counter += quantity;
+    });
+
+    const avg = total / counter;
+    const round = (Math.round(avg * 4) / 4).toFixed(2);
+    setRating(round);
   };
 
-  const getMetadata = () => {
-    if (currentProduct) {
-      axios.get(`/api/reviews/meta/${currentProduct.id}`)
+  const getMetadata = (id, selected) => {
+    if (relatedItem) {
+      axios.get(`/api/reviews/meta/${id}`)
         .then((results) => {
           getRating(results.data.ratings);
+          if (!selected) {
+            setRelatedChars(results.data.characteristics);
+          } else {
+            setSelectedChars(results.data.characteristics);
+          }
         })
         .catch((err) => {
           console.log('err getting metadata', err);
```

Comment:
Same stuff here. Could re-factor the app.js getMetadata function to take in parameters and maybe a callback function to handle the response data. Or use promises too.